File: 08_03_2025_Minimum_Recolors_to_Get_K_Consecutive_BlackBlocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int left=0,now=0,nr=INT_MAX;
        for(int right=0;right<blocks.size();right++){
            if(blocks[right]=='W')
            now++;
            if(right-left+1==k){
                nr=min(nr,now);
                if(blocks[left]=='W')
                now--;
                left++;
            }
        }
        return nr;
    }
};

File: 08_03_2025_Minimum_Recolors_to_Get_K_Consecutive_BlackBlocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int left = 0, now = 0, nr = Integer.MAX_VALUE;
        
        for (int right = 0; right < blocks.length(); right++) {
            if (blocks.charAt(right) == 'W') {
                now++;
            }
            if (right - left + 1 == k) {
                nr = Math.min(nr, now);
                if (blocks.charAt(left) == 'W') {
                    now--;
                }
                left++;
            }
        }
        return nr;
    }
}
